modin/engines/base/frame/data.py

```diff
@@ -24,9 +30,26 @@
 
 
 class BasePandasFrame(object):
-    """An abstract class that represents the Parent class for any Pandas DataFrame class.
-
-    This class is intended to simplify the way that operations are performed
+    """An abstract class that represents the parent class for any Pandas backend dataframe class.
+
+    This class provides interfaces to run operations on dataframe partitions.
+
+    Parameters
```

Comment:
This should be under init.

Reply:
constructor `__init__` should be documented in class-docstring according to [nympydoc](https://numpydoc.readthedocs.io/en/latest/format.html#documenting-classes)

Reply:
The script mentioned in #2934 says the following: `__init__:GL08:The object does not have a docstring`. That issue needs to be clarified. cc @anmyachev

Reply:
Good example: https://github.com/numpy/numpy/blob/v1.20.0/numpy/ma/core.py#L2702-L6217

> constructor __init__ should be documented in class-docstring according to numpydoc

In `__init__` method, we can write a couple of words and refer to class docstring. Some of the developers may not know all the details of `numpydoc` standard and, if necessary, may not find the right place.

Reply:
So, let's put a docstring as the guide says, i.e. at the level of a class. But we need to correct the script because currently it dumps `__init__:GL08:The object does not have a docstring`.

Reply:
I suggest to write a short docstring for `__init__` (skipping parameters description, etc.)

Reply:
I don't think it is necessary.

Reply:
Docstring for method `__init__` is [optional](https://github.com/modin-project/modin/pull/2968#issuecomment-821206140) now